src/tests/test_parse.py

```diff
@@ -298,28 +299,73 @@ def complete_with_defaults(mapping):
 
 
 @pytest.mark.parametrize(
-    "line, expected",
+    "line, cmd_from, expected_error_message",
     [
         (
-            "some-argument",
-            {"line": ["some-argument"]},
+            "duckdb:// --alias test1 --alias test2",
+            "sql",
+            (
+                "Duplicate arguments in %sql. "
+                "Please use only one of each of the following: --alias"
+            ),
         ),
         (
-            "a b c",
-            {"line": ["a", "b", "c"]},
+            """histogram --table penguins.csv --column bill_length_mm
+--column body_mass_g""",
+            "sqlplot",
+            (
+                "Duplicate arguments in %sqlplot. "
+                "Please use only one of each of the following: --column"
+            ),
         ),
+        (
+            """bar --table penguins.csv --column bill_length_mm
+--show-numbers --show-numbers""",
+            "sqlplot",
+            (
+                "Duplicate arguments in %sqlplot. "
+                "Please use only one of each of the following: --show-numbers"
+            ),
+        ),
+    ],
+)
+def test_magic_args_raises_usageerror(
+    load_penguin, ip, line, cmd_from, expected_error_message
+):
+    ALLOWED_DUPLICATES = {
+        "sql": ["-w", "--with", "--append", "--interact"],
+        "sqlplot": ["-w", "--with"],
+        "sqlcmd": [],
+    }
+
+    sql_line = ip.magics_manager.lsmagic()["line"][cmd_from]
+
+    with pytest.raises(UsageError) as excinfo:
+        magic_args(
+            sql_line,
+            line,
+            cmd_from,
+            ALLOWED_DUPLICATES[cmd_from],
+        )
+    assert expected_error_message in str(excinfo.value)
```

Comment:
let's test this by calling the actual `%sql` (and others) magic

Reply:
Fixed [here](https://github.com/ploomber/jupysql/pull/908/files#diff-edd543175e806ac6654196270596c90240cc3a3cfdcba20d35d997207e4555f0R303-R618)